### ISSUE TITLE
Finding reported on wrong line when inside macro

### ISSUE THREAD

**Issue by peckto** [Ericsson/codechecker#4478]
**Describe the bug**
When a finding is within a macro, the finding is reported at the line of the macro and not at the line of the actual statement.
This has been observed on the clangsa analyzer, not sure if this is a problem related to clangsa or to CodeChecker.

**CodeChecker version**
```
$ CodeChecker  version
[INFO 2025-03-04 13:56] - CodeChecker analyzer version:
---------------------------------------------------------------
Kind                 | Version                                 
---------------------------------------------------------------
Base package version | 6.24.4                                  
Package build date   | 2024-10-28T15:46                        
Git commit ID (hash) | 454d978191ed42c8202a2244dc092dfb6fd8c83a
Git tag information  | 6.24.4                                  
---------------------------------------------------------------

[INFO 2025-03-04 13:56] - CodeChecker web version:
------------------------------------------------------------------------------
Kind                                | Version                                 
------------------------------------------------------------------------------
Base package version                | 6.24.4                                  
Package build date                  | 2024-10-28T15:46                        
Git commit ID (hash)                | 454d978191ed42c8202a2244dc092dfb6fd8c83a
Git tag information                 | 6.24.4                                  
Server supported Thrift API version | 6.58                                    
Client Thrift API version           | 6.58                                    
------------------------------------------------------------------------------
```

**To Reproduce**
`poc.c`
```c
#include <stdio.h>
#include <string.h>

int static state = 1;

#define IF_STATE_1(CODE) \
    if (state == 1) { \
        CODE; \
    }\

int main() {
    char buf[10] = {0};

    IF_STATE_1(
        printf("foo\n");
        memset(buf, 1, 11);
    )
    return 0;
}
```

```
$ bear -- clang -c poc.c
$ analyze -o report --analyzers clangsa --enable-all compile_commands.json
$ CodeChecker parse -e html -o report-html report
$ CodeChecker parse report
[MEDIUM] poc.c:14:5: Call to function 'memset' is insecure as it does not provide security checks introduced in the C11 standard. Replace with analogous functions that support length arguments or provides boundary checks such as 'memset_s' in case of C11 [security.insecureAPI.DeprecatedOrUnsafeBufferHandling]
    IF_STATE_1(
    ^
```

![Image](https://github.com/user-attachments/assets/5c8f35f2-47e4-4074-8838-d3934a696fc4)

**Expected behaviour**
Finding should be reported at line 16 (`memset(buf, 1, 11)`).

**Desktop (please complete the following information)**
 - OS: Linux
 - Version: Fedora 41

**Additional context**
N/A
